src/server/responder.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <atomic>
+#include <memory>
+
+#include "absl/types/span.h"
+#include "base/logging.h"
+
+namespace dfly {
+
+class ConnectionContext;
+
+class Responder {
+ public:
+  virtual ~Responder() = default;
+
+  virtual bool Wait() = 0;  // Return for result portion, returns true when done.
+
+  virtual void Respond(ConnectionContext* cntx) = 0;  // Send result portion.
+};
+
+constexpr int kResponderSizeLimit = 64;
+
+template <typename RT, typename... Args> RT* MakeResponder(absl::Span<char> buf, Args&&... args) {
```

Comment:
I think SMO is too early since it works for a single command. See my question about pipelining but i can ask the same about multi/exec.